lib/spack/spack/package.py

```diff
@@ -2637,7 +2640,16 @@ def test_process(pkg, kwargs):
                     if not isinstance(test_fn, types.FunctionType):
                         test_fn = test_fn.__func__
 
+                    # Skip any test methods consisting solely of 'pass'
+                    # since they do not contribute to package testing.
+                    source = (inspect.getsource(test_fn)).split('\n')[1:]
+                    lines = [ln.strip() for ln in source]
```

Comment:
Minor "best practices" suggestion: use a generator expression for this instead of forcing construction of a list
```suggestion
                    lines = (ln.strip() for ln in source)
```